misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractCharSequenceExpectationsTest.kt

```diff
@@ -33,6 +37,24 @@ abstract class AbstractCharSequenceExpectationsTest(
         notToMatchSpec.forSubjectLessTest(Regex(""))
     )
 
+    @TestFactory
+    fun length_feature_and_function() = testFactoryForFeatureNonFeature(
+        lengthFeatureSpec,
+        lengthSpec
+    ) { name, lengthFun, hasExtraHint ->
+        it("$name - passes" + showsSubExpectationIf(hasExtraHint)) {
+            expect("Hello" as CharSequence).lengthFun { toBeLessThan(30) }
+        }
+        it("$name - fails" + showsSubExpectationIf(hasExtraHint)) {
+            expect {
+                expect("Hi" as CharSequence).lengthFun { toBeLessThan(2) }
+            }.toThrow<AssertionError> {
+                messageToContain(toBeLessThanDescr)
+                if (hasExtraHint) messageToContain("length")
+            }
```

Comment:
My bad, since the length extraction always succeeds there is no difference regarding failure hints
```suggestion
        it("$name - fails")) {
            expect {
                expect("Hi" as CharSequence).lengthFun { toBeLessThan(2) }
            }.toThrow<AssertionError> {
                messageToContain("length: 2", "$toBeLessThanDescr: 2")
            }
```